include/seqan3/core/type_traits/iterator.hpp

```diff
@@ -231,3 +231,45 @@ using iterator_tag_t = typename iterator_tag<it_t>::type;
 //!\}
 
 } // namespace seqan3
+
+namespace seqan3::detail
+{
+// ----------------------------------------------------------------------------
+// iter_pointer
+// ----------------------------------------------------------------------------
+
+/*!\brief This is like std::iter_value_t, but for the pointer type.
+ * \implements seqan3::transformation_trait
+ * \tparam it_t The type to operate on.
+ * \see seqan3::detail::iter_pointer_t
+ *
+ * \attention
+ * C++20 does not provide a `std::iter_pointer_t`, because the new C++20 iterators do not need to provide a pointer
+ * type.
+ */
+template <typename it_t>
+struct iter_pointer
```

Comment:
Why don't you call it `pointer_type`?

Reply:
In C++20 they follow the following name scheme for member of [`iterator_traits`](https://en.cppreference.com/w/cpp/iterator/iterator_traits):

* `std::iterator_traits::difference_type` -> `std::iter_difference_t`
* `std::iterator_traits::value_type` -> `std::iter_value_t`
* `std::iterator_traits::pointer` -> `seqan3::detail::iter_pointer_t`
* `std::iterator_traits::reference` -> `std::iter_reference_t`
* `std::iterator_traits::iterator_category` -> ???

Generally in C++ a type `<name>_t` is a shortcut for `using <name>_t = typename <name>::type;`

Since the shortcut is named `seqan3::detail::iter_pointer_t`, the struct should be named `seqan3::detail::iter_pointer`.

Reply:
Thanks for the explanation!
I were just looking at `std::iter_value_t`, where it's called `value_type`.